libsignal-service/src/account_manager.rs

```diff
@@ -42,4 +52,57 @@ impl<Service: PushService> AccountManager<Service> {
             .confirm_device(confirmation_code, confirm_device_message)
             .await?)
     }
+
+    /// Checks the availability of pre-keys, and updates them as necessary.
+    ///
+    /// Parameters are the protocol's `StoreContext`, and the offsets for the next pre-key and
+    /// signed pre-keys.
+    ///
+    /// Equivalent to Java's RefreshPreKeysJob
+    pub async fn update_pre_key_bundle(
+        &mut self,
+        store_context: StoreContext,
+        pre_keys_offset_id: u32,
```

Comment:
What about outputting the new value (so the caller doesn't have to know how many pre-keys we generate, if we did that etc.) as either a return value, or taking the parameter as `&mut pre_keys_offset_id`?

Reply:
Good idea. Whisperfish (pre 0.5) used a random u32 as next pre-key ID, and I just look at those that I have stored and go from there. But returning it is very useful indeed.